embedded-hal-async/src/digital.rs

```diff
@@ -30,7 +30,7 @@ pub trait Wait: embedded_hal::digital::ErrorType {
     /// # Note for implementers
     /// The pin may have switched back to low before the task was run after
     /// being woken. The future should still resolve in that case.
-    fn wait_for_high<'a>(&'a mut self) -> Self::WaitForHighFuture<'a>;
+    fn wait_for_high(&mut self) -> Self::WaitForHighFuture<'_>;
```

Comment:
Do these changes need an entry in the changelog?

Reply:
No, they're equivalent (lifetime elision). Clippy wants the elided version if possible. Anyway this is now split to #406.